src/Module/Settings/Uexport.php

```diff
@@ -0,0 +1,207 @@
+
+		$tpl = Renderer::getMarkupTemplate("uexport.tpl");
+		return Renderer::replaceMacros($tpl, [
+			'$title' => L10n::t('Export personal data'),
+			'$options' => $options
+		]);
+	}
+	private function uexport_multirow($query) {
+		global $dbStructure;
+
+		preg_match("/\s+from\s+`?([a-z\d_]+)`?/i", $query, $match);
+		$table = $match[1];
+
+		$result = [];
+		$r = q($query);
+		if (DBA::isResult($r)) {
+			foreach ($r as $rr) {
+				$p = [];
+				foreach ($rr as $k => $v) {
+					switch ($dbStructure[$table]['fields'][$k]['type']) {
+						case 'datetime':
+							$p[$k] = $v ?? DBA::NULL_DATETIME;
+							break;
+						default:
+							$p[$k] = $v;
+							break;
+					}
+				}
+				$result[] = $p;
+			}
+		}
+		return $result;
+	}
+
+	private function uexport_row($query) {
+		global $dbStructure;
+
+		preg_match("/\s+from\s+`?([a-z\d_]+)`?/i", $query, $match);
+		$table = $match[1];
+
+		$result = [];
+		$r = q($query);
+		if (DBA::isResult($r)) {
+
+			foreach ($r as $rr) {
+				foreach ($rr as $k => $v) {
+					switch ($dbStructure[$table]['fields'][$k]['type']) {
+						case 'datetime':
+							$result[$k] = $v ?? DBA::NULL_DATETIME;
+							break;
+						default:
+							$result[$k] = $v;
+							break;
+					}
+				}
+			}
+		}
+		return $result;
+	}
+
+	private function uexport_account($a) {
+
+		$user = SELF::uexport_row(
+			sprintf("SELECT * FROM `user` WHERE `uid` = %d LIMIT 1", intval(local_user()))
+		);
+
+		$contact = SELF::uexport_multirow(
+			sprintf("SELECT * FROM `contact` WHERE `uid` = %d ", intval(local_user()))
+		);
+
+
+		$profile = SELF::uexport_multirow(
+			sprintf("SELECT * FROM `profile` WHERE `uid` = %d ", intval(local_user()))
+		);
+
+		$photo = SELF::uexport_multirow(
+			sprintf("SELECT * FROM `photo` WHERE uid = %d AND profile = 1", intval(local_user()))
+		);
+		foreach ($photo as &$p) {
+			$p['data'] = bin2hex($p['data']);
+		}
+
+		$pconfig = SELF::uexport_multirow(
+			sprintf("SELECT * FROM `pconfig` WHERE uid = %d", intval(local_user()))
+		);
+
+		$group = SELF::uexport_multirow(
+			sprintf("SELECT * FROM `group` WHERE uid = %d", intval(local_user()))
+		);
+
+		$group_member = SELF::uexport_multirow(
+			sprintf("SELECT `group_member`.`gid`, `group_member`.`contact-id` FROM `group_member` INNER JOIN `group` ON `group`.`id` = `group_member`.`gid` WHERE `group`.`uid` = %d", intval(local_user()))
+		);
+
+		$output = [
+			'version' => FRIENDICA_VERSION,
+			'schema' => DB_UPDATE_VERSION,
+			'baseurl' => System::baseUrl(),
+			'user' => $user,
+			'contact' => $contact,
+			'profile' => $profile,
+			'photo' => $photo,
+			'pconfig' => $pconfig,
+			'group' => $group,
+			'group_member' => $group_member,
+		];
+
+		echo json_encode($output, JSON_PARTIAL_OUTPUT_ON_ERROR);
+	}
+
+	/**
+	 * echoes account data and items as separated json, one per line
+	 *
+	 * @param App $a
+	 * @throws Exception
+	 */
+	private function uexport_all(App $a) {
+
+		SELF::uexport_account($a);
+		echo "\n";
```

Comment:
Is this echo intended? If so, it should be part of the template

Reply:
Yes it is, that way the profile data is separated from the content data by the newline and one can get the profile data for uimport should one only has the full backup. The JSON file is directly handed over to the browser, which shows the user the save / open dialog or executes the default action for the file type.